Python/symtable.c

```diff
@@ -1107,6 +1158,70 @@ symtable_add_def(struct symtable *st, PyObject *name, int flag,
                         lineno, col_offset, end_lineno, end_col_offset);
 }
 
+static int
+symtable_enter_typeparam_block(struct symtable *st, identifier name,
+                               void *ast, int has_defaults, int has_kwdefaults,
+                               enum _stmt_kind kind,
+                               int lineno, int col_offset,
+                               int end_lineno, int end_col_offset)
+{
+    _Py_block_ty current_type = st->st_cur->ste_type;
+    if(!symtable_enter_block(st, name, TypeParamBlock, ast, lineno,
+                             col_offset, end_lineno, end_col_offset)) {
+        return 0;
+    }
+    if (current_type == ClassBlock) {
+        st->st_cur->ste_type_params_in_class = 1;
+        _Py_DECLARE_STR(namespace, ".namespace");
```

Comment:
TODO what is this for?

Reply:
We use them to store some data in the locals that needs to be moved around during the runtime. I think this particular one might no longer be used; I'll double-check.

Reply:
It is indeed no longer used, but I may need to bring it back in order to simplify LOAD_LOCALS.

Reply:
Put it back as part of the effort to simplify LOAD_LOCALS.